vertical-pod-autoscaler/pkg/utils/vpa/api.go

```diff
@@ -49,32 +50,34 @@ func patchVpa(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string,
 	return vpaClient.Patch(vpaName, types.JSONPatchType, bytes)
 }
 
-// InitVpaStatus inserts into VPA CRD object an empty VerticalPodAutoscalerStatus object, so later it can be filled with data.
-func InitVpaStatus(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string) (result *vpa_types.VerticalPodAutoscaler, err error) {
-	return patchVpa(vpaClient, vpaName, []patchRecord{{
-		Op:    "add",
-		Path:  "/status",
-		Value: vpa_types.VerticalPodAutoscalerStatus{},
-	},
-	})
-}
+// UpdateVpaStatus updates the status field of the VPA API object.
+// It prevents race conditions by verifying that the lastUpdateTime of the
+// API object and its model representation are equal.
+func UpdateVpaStatus(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpa *model.Vpa) (result *vpa_types.VerticalPodAutoscaler, err error) {
+	status := vpa_types.VerticalPodAutoscalerStatus{
+		LastUpdateTime: metav1.Now(),
+		Conditions:     vpa.Conditions.AsList(),
+	}
+	if vpa.Recommendation != nil {
+		status.Recommendation = *vpa.Recommendation
+	}
 
-// UpdateVpaRecommendation updates VPA's status/recommendation object and status/lastUpdateTime.
-func UpdateVpaRecommendation(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string, recommendation vpa_types.RecommendedPodResources) (result *vpa_types.VerticalPodAutoscaler, err error) {
-	patches := []patchRecord{
-		{
-			Op:    "add",
+	patches := make([]patchRecord, 0)
+	if !vpa.LastUpdateTime.IsZero() {
```

Comment:
Why not test in that case? There's a race for setting the first status then, no?

Reply:
Unfortunately I cannot set a test on a missing field in the patch method.
On the other hand we can make the assumption that the Updater (or other agents) don't do any operations on a VPA that doesn't have the Status field. In particular it has no recommendation at this point, so the Updater can't do anything useful. WDYT?

Reply:
Done (discussed offline, if the lastUpdateTime is missing I set a test on the whole status field).